lib/server/ServerResponse.js

```diff
@@ -35,6 +53,17 @@ var ServerResponse = Response.extend({
     }
 });
 
+function addAllFields(passedHeaders, headers) {
+    var fields = Object.keys(passedHeaders);
+    var currentField;
+
+    while(fields.length) {
+        currentField = fields.shift();
+
+        headers[currentField] = passedHeaders[currentField];
+    }
```

Comment:
This is O(n^2) since shift is O(n). It would be faster to do pop than shift. Any reason why you chose to mutate fields anyway? Why not just do `Object.keys(passedHeaders).forEach(function(key) { ... });`?